DSESwift/DataStorageEngine.swift

```diff
@@ -7,40 +7,137 @@
 //
 
 import Foundation
+import PromiseKit
 
-enum DatabaseError: ErrorType{
-    case ObjectNotFound
-    case InsertFailed
-    case UpdateFailed
-    case DuplicatedObjects
-    case InvalidLimit
-    case UnauthenticatedAccess
-    case UnknownError
+public enum DatabaseError: ErrorType{
+    case InvalidAppKey
+    case InvalidSessionId
+    case InvalidUserId
+    case EmptyCredentials
 }
```

Comment:
Yes, you can cast it back to an specific error from ErrorType. But DSE has 3 different errors atm, so I will clean it up.